app/javascript/src/utils/codemirror/indent.js

```diff
@@ -113,8 +115,9 @@ export function autoIndentOnEnter({ state, dispatch }) {
   const changes = state.changeByRange(range => {
     const { from, to } = range, line = state.doc.lineAt(from)
 
-    let indent = getIndentForLine(state, from, from - line.from)
-    if (shouldNextLineBeIndent(line.text)) indent++
+    const cursorPos = from - line.from
+    let indent = getIndentForLine(state, from, cursorPos)
+    if (shouldNextLineBeIndent(line.text.slice(0, cursorPos))) indent++
```

Comment:
Perhaps we should always use the cursor position, I don't see a scenario where we would want to use the end of the line instead, but I could be wrong.

In any case, let's not use abbreviations

```suggestion
    const cursorPosition = from - line.from
    let indent = getIndentForLine(state, from, cursorPosition)
    if (shouldNextLineBeIndent(line.text.slice(0, cursorPosition))) indent++
```

Reply:
the way it works right now it uses array slice at the cursor. i guess if we ever want to look at anything past the cursor its no good, but i dont think we do? either way i didnt want to mess with the parts of the code that already work 😅

Reply:
Sure, fair enough